python/beeai_framework/utils/asynchronous.py

```diff
@@ -40,3 +40,33 @@ async def wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
 async def to_async_generator(items: list[T]) -> AsyncGenerator[T]:
     for item in items:
         yield item
+
+
+def awaitable_to_coroutine(awaitable: Awaitable[T]) -> Coroutine[Any, Any, T]:
+    async def as_coroutine() -> T:
+        return await awaitable
+
+    return as_coroutine()
+
+
+def run_sync(awaitable: Awaitable[T], *, timeout: int | None = None) -> T:
+    """
+    Run *awaitable* from synchronous code.
+
+    - If we're already inside the loop's thread, raise an error (to avoid dead-lock).
+    - If no loop is running, create one temporarily.
+    - If a loop is running in another thread, schedule thread-safely.
+    """
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        if asyncio.iscoroutine(awaitable):
+            return asyncio.run(awaitable, debug=False)
+        else:
+            return asyncio.run(awaitable_to_coroutine(awaitable), debug=False)
+
+    if loop.is_running() and loop == asyncio.get_running_loop():
+        raise RuntimeError("blocking_await() called from inside the event-loop thread; would dead-lock")
```

Comment:
This check should be after `loop = asyncio.get_running_loop()`. Here it is redundant.